packages/blade/src/components/Tabs/_decisions/decisions.md

```diff
@@ -0,0 +1,385 @@
+# Tabs API Decisions <!-- omit in toc -->
+
+A tab is a navigation component used in the interface to switch between different views in the same context. Tabs are contextual to the section or the page and are triggered by user interaction.
+
+<img src="./tabs-thumbnail.png" width="380" alt="Tabs thumbnail" />
+
+- [Design](#design)
+- [Features](#features)
+- [Anatomy](#anatomy)
+- [`Tabs` API](#tabs-api)
+- [`TabItem` API](#tabitem-api)
+- [`TabPanel` API](#tabpanel-api)
+  - [Examples:](#examples)
+- [Basic](#basic)
+    - [Controlled, Uncontrolled](#controlled-uncontrolled)
+- [Vertical Orientation](#vertical-orientation)
+- [Filled Variant](#filled-variant)
+- [With Tooltip](#with-tooltip)
+- [Alternative APIs](#alternative-apis)
+  - [1. Quite tricky/hack to implement](#1-quite-trickyhack-to-implement)
+  - [2. The JSX doesn't map well with the visual hierarchy / the HTML structure](#2-the-jsx-doesnt-map-well-with-the-visual-hierarchy--the-html-structure)
+  - [3. Extensibility](#3-extensibility)
+  - [4. API Familiarity](#4-api-familiarity)
+- [Motion](#motion)
+- [Accessibility](#accessibility)
+- [Open Questions](#open-questions)
+
+## Design
+
+[Figma Link](https://www.figma.com/file/LSG77hEeVYDk7j7WV7OMJE/Blade-DSL---Components-Guideline?node-id=3733%3A8981&mode=dev) to all variants of the Tabs component
+
+## Features
+
+- Controlled and Uncontrolled
+- Filled and Bordered variants
+- Horizontal and Vertical orientation
+- Medium and Large size
+
+## Anatomy
+
+<img src="./tabs-anatomy.png" width="100%" alt="Tabs thumbnail" />
+
+```jsx
+<Tabs>
+  <TabList>
+    <TabItem />
+  </TabList>
+  <TabPanel />
+</Tabs>
+```
+
+## `Tabs` API
+
+```ts
+type TabsProps = {
+  /**
+   * The content of the component, accepts `TabList` and `TabPanel` components.
+   */
+  children: React.ReactNode;
+  /**
+   * The value of the selected tab, If set the component will be controlled.
+   */
+  value?: string;
+  /**
+   * The default value of the selected tab, in case the Tabs component is uncontrolled.
+   */
+  defaultValue?: string;
+  /**
+   * Callback fired when the value changes.
+   */
+  onChange?: (value: string) => void;
+  /**
+   * The orientation of the tabs.
+   */
+  orientation?: 'horizontal' | 'vertical';
+  /**
+   * The size of the tabs.
+   */
+  size?: 'medium' | 'large';
+  /**
+   * The variant of the tabs.
+   */
+  variant?: 'bordered' | 'filled';
+  /**
+   * If `true`, the TabItems will grow to use all the available space.
+   */
+  autoWidth?: boolean;
```

Comment:
Okay. So in this case the `TabItems` would stretch to occupy the whole `Tabs`, right?

Reply:
Yeah check the design [here](https://www.figma.com/file/LSG77hEeVYDk7j7WV7OMJE/Blade-DSL---Components-Guideline?node-id=3733%3A10954&mode=dev)

<img width="648" alt="image" src="https://github.com/razorpay/blade/assets/35374649/a8412e90-5e4b-4487-8f5f-c86c5e9177f6">

Reply:
autoWidth is confusing too. because i'm unable to understand what will it do? will it adjust the width of the tab or tablist or tab panel as it affects the width of the tabList. can we name it something like `isFullWidthTabList`?

Reply:
isFullWidthTabList, too verbose. 

I've seen design systems use "isFitted" prop. which might be better?

Reply:
again adds a learning curve. i'm okay with verbose name as far as it conveys the purpose without me fiddling around and guessing what will it do.